### ISSUE TITLE
'ERROR: missing FROM-clause entry for table' when querying nested nodes

### ISSUE THREAD

**Issue by Jaszkowic** [prisma/prisma#2954]
**Describe the bug**

Querying nested nodes results in error

> org.postgresql.util.PSQLException: ERROR: missing FROM-clause entry for table "Session_TaskGroup_Task_Alias"

**To Reproduce**

1. Setup Prisma 1.14.2 with latest PostgreSQL database

2. Schema:

```graphql
type Session {
    id: ID! @unique,
    taskGroups: [TaskGroup!]! @relation(name: "TaskGroupsForSession", onDelete: CASCADE)
}

type TaskGroup {
    id: ID! @unique,
    session: Session!  @relation(name: "TaskGroupsForSession"),
    tasks: [Task!]! @relation(name: "TasksForTaskGroups", onDelete: CASCADE)
}

type Task {
    id: ID! @unique,
    taskGroup: TaskGroup! @relation(name: "TasksForTaskGroups"),
    template: Template @relation(name: "TemplateForTask", onDelete: CASCADE)
}

type Template {
    id: ID! @unique,
    task: Task @relation(name: "TemplateForTask")
}
```

3. Seed with some data

4. Query

This works:

```graphql
query templates(where:
  task: {
    taskGroup: {
      id: "abc"
    }
  }
){
  id
}
```

But this doesn't:

```graphql
query templates(where:
  task: {
    taskGroup: {
      session: {
        id: "xyz"
      }
    }
  }
){
  id
}
```

**Expected behavior**
Query should succeed.

**Versions (please complete the following information):**
Prisma 1.14.2 with latest PostgreSQL database

<details>
  <summary>** Error output **</summary>
  > [36mprisma_1    |[0m [Bugsnag - local / testing] Error report: com.bugsnag.Report@646c63d8
[36mprisma_1    |[0m org.postgresql.util.PSQLException: ERROR: missing FROM-clause entry for table "Session_TaskGroup_Task_Alias"
[36mprisma_1    |[0m   Position: 493
[36mprisma_1    |[0m 	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2433)
[36mprisma_1    |[0m 	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2178)
[36mprisma_1    |[0m 	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:306)
[36mprisma_1    |[0m 	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:441)
[36mprisma_1    |[0m 	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:365)
[36mprisma_1    |[0m 	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:155)
[36mprisma_1    |[0m 	at org.postgresql.jdbc.PgPreparedStatement.executeQuery(PgPreparedStatement.java:118)
[36mprisma_1    |[0m 	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeQuery(ProxyPreparedStatement.java:52)
[36mprisma_1    |[0m 	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeQuery(HikariProxyPreparedStatement.java)
[36mprisma_1    |[0m 	at com.prisma.api.connector.jdbc.database.BuilderBase.$anonfun$queryToDBIO$1(BuilderBase.scala:51)
[36mprisma_1    |[0m 	at com.prisma.api.connector.jdbc.database.BuilderBase.$anonfun$jooqToDBIO$1(BuilderBase.scala:81)
[36mprisma_1    |[0m 	at slick.jdbc.SimpleJdbcAction.run(StreamingInvokerAction.scala:70)
[36mprisma_1    |[0m 	at slick.jdbc.SimpleJdbcAction.run(StreamingInvokerAction.scala:69)
[36mprisma_1    |[0m 	at slick.basic.BasicBackend$DatabaseDef$$anon$2.liftedTree1$1(BasicBackend.scala:275)
[36mprisma_1    |[0m 	at slick.basic.BasicBackend$DatabaseDef$$anon$2.run(BasicBackend.scala:275)
[36mprisma_1    |[0m 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[36mprisma_1    |[0m 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[36mprisma_1    |[0m 	at java.lang.Thread.run(Thread.java:748)

</details>

**Comment by joefru** [comment]
@Jaszkowic Were you able to find a workaround? This issue is currently preventing me from upgrading to 1.14+.

**Comment by Jaszkowic** [comment]
@joefru No, unfortunately not. I downgraded to 1.12.0 for now. But i think it would work to perform the nested query one by one, e.g. query task, then query taskGroup from task, then query session from taskGroup etc. But it would involve a lot of manual filtering.

**Comment by joefru** [comment]
To me, this issue is critical. How is anyone with more than the simplest of applications using version 1.14+? All those performance improvements... want them sooo bad! 😄

**Comment by mavilein** [comment]
Thanks for reporting and providing thorough descriptions! 🙏  We fixed this issue and it is available in version `1.15.2` 🎉 . The build will be available in a few minutes.
